website/static/js/filepage/index.js

```diff
@@ -363,6 +363,64 @@ var FileViewPage = {
         }, 1000);
 
         self.mfrIframeParent = $('#mfrIframeParent');
+        function goToRevisions(e){
+          var editable = self.editor && self.editor.selected;
+          var viewable = self.mfrIframeParent.is(':visible');
+          if (editable || viewable){
+              if (viewable){
+                  self.mfrIframeParent.toggle();
+              }
+              if (editable) {
+                  self.editor.selected = false;
+              }
+          }
+          self.revisions.selected = true;
+        }
+
+        function changeVersionHeader(value){
+          m.render(document.getElementById('version-link'), m('a', {onclick: goToRevisions},'(Version:  ' + String(value) + ')'));
+        }
+
+        function httpGetAsync(theUrl, callback) {
+            var xmlHttp = new XMLHttpRequest();
+            xmlHttp.onreadystatechange = function() {
+                if (xmlHttp.readyState === 4 && xmlHttp.status === 200)
+                    callback(xmlHttp.responseText);
+            };
+            xmlHttp.open('GET', theUrl, true); // true for asynchronous
+            xmlHttp.send(null);
+        }
+
+        function setVersionHeader(responseText) {
+            var obj = JSON.parse(responseText);
+            changeVersionHeader(obj.data[0].version);
+        }
+
+        //anchor checking hack that will select if true
+        var idx = window.location.href.indexOf('?');
```

Comment:
Check out URI.js - it should be better than string manipulation and we already use it in our stack. https://medialize.github.io/URI.js/